Mage/src/main/java/mage/target/TargetSpell.java

```diff
@@ -108,6 +108,7 @@ public TargetSpell copy() {
 
     private boolean canBeChosen(StackObject stackObject, UUID sourceControllerId, Ability source, Game game) {
         return stackObject instanceof Spell
+                && !stackObject.getId().equals(source.getId()) // 115.5. A spell or ability on the stack is an illegal target for itself.
                 && game.getState().getPlayersInRange(sourceControllerId, game).contains(stackObject.getControllerId())
                 && canTarget(sourceControllerId, stackObject.getId(), source, game);
```

Comment:
canTarget already check it:

![IMG_0874](https://github.com/user-attachments/assets/a187f0d3-9a8e-403e-b162-f9c20fc193e9)

Reply:
Hmm, old code never works? Looks like it used wrong target id to compare:

![IMG_0875](https://github.com/user-attachments/assets/01b1a60e-4e0c-4169-8ec5-7f9bbfc8217d)

Reply:
nvm

Reply:
But code place is important anyway. It must check itself on canTarget, so possible targets and override targets usage will be compatible too.